### PR TITLE
Apply minor changes to action buttons in data collection flow

### LINKED ISSUE #1980: Incorrect buttons and order on "take a photo" task
After taking a photo, current buttons are:

"Skip, Continue, Undo icon"

The should be:

"Undo icon, Continue".

i.e. The icon should be first, and the Skip button should not be shown when there's a photo present.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1980

[take_a_photo_button_fixes.webm](https://github.com/google/ground-android/assets/8918023/78105b91-892d-4950-8959-a504ae836bec)

This screencast was taken before renaming the button to "next"

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt [hunk 1]
```diff
@@ -123,9 +124,13 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
       .setOnTaskUpdated { button, taskData -> button.enableIfTrue(taskData.isNotNullOrEmpty()) }
       .disable()
 
+  /** Skip button is only visible iff, the task is optional and the task data is missing. */
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:127]: Nit: Extra comma after "iff".

**Review by gino-m — CHANGES_REQUESTED** [review]
LGTM! Can we also update the button label from "Continue" to "Next" as per latest mocks while you're at it?

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt]: Done

**Comment by shobhitagarwal1612** [comment]
> Can we also update the button label from "Continue" to "Next" as per latest mocks while you're at it?

Done

**Review by gino-m — APPROVED** [review]
LGTM

**Pull request merged**